src/PublishCodeCovCoverageTask/utils/webUtils.ts

```
@@ -10,208 +10,252 @@ import * as url from 'node:url';
+                if (!signal.aborted) {
+                    // Only abort our own controller - external signals are managed by the caller
+                    if (signal === controller.signal) {
+                        controller.abort();
                     }
+                }
 
-                    // Check if we've exceeded the maximum number of redirects
-                    if (redirectCount >= maxRedirects) {
-                        return reject(new Error(`Maximum redirect count (${maxRedirects}) reached for '${fileUrl}'`));
+                // Destroy the response stream if it exists
+                if (response && response.data) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)